### PR TITLE
Updated Docs engine

### LINKED ISSUE #157: prismjs syntax highllighting
Would be nice to have syntax highlighting for a `cml` language based on https://github.com/iterative/dvc.org/blob/master/config/prismjs/dvc.js

- related #66

### LINKED ISSUE #187: ref: auto-link options
Port over https://github.com/iterative/dvc.org/pull/3140 after #186

### LINKED ISSUE #238: Add header and footer to cml.dev 404 page

### PR DESCRIPTION
**Pull request by yathomasi**
Updated docs engine to use reusable package [gatsby-theme-iterative](https://github.com/iterative/gatsby-theme-iterative/).
The latest `gatsby-theme-iterative@0.1.0` update https://github.com/iterative/gatsby-theme-iterative/pull/7 does include features requested for cml docs. 

Fixes #157
Fixes #187 
Fixes #238 

A Example usage: 
https://cmldev-docsengine.gtsb.io/doc/ref/ci
https://github.com/iterative/cml.dev/pull/233/commits/0ca4d401bd5cc4424481c2096bcdc0d326f2b00f#diff-2285ff562ca51e66d2c3fc4bc92b4a380b47afab4449dfd2c5a07c221f9d9305

### CODE REVIEW

File: src/gatsby/models.js [hunk 1]
```diff
@@ -1,9 +1,8 @@
 const markdownContent = require('./models/markdown-content')
-const docs = require('./models/docs')
 const imageSourcePaths = require('./models/image-source-paths')
 const github = require('./models/github')
 const pruneCache = require('./models/prune-cache')
 
-const models = [markdownContent, docs, imageSourcePaths, github, pruneCache]
+const models = [markdownContent, imageSourcePaths, github, pruneCache]
```
Comment (rogermparent) [inline src/gatsby/models.js:6]: I think `markdownContent` can be removed as well, since nothing aside from docs used it as cml.dev has no blog

**Review by rogermparent — COMMENTED** [review]
Looking good so far! Awesome that it fixes some existing bugs. There could be more scrubbing done to remove what the theme replaces, but that's not a hugely pressing issue. 

I did notice [the 404 page](https://cmldev-docsengine.gtsb.io/404) doesn't have headers or footers though, which would be annoying in specific situations where the user hits 404 without a page to go back to (outdated links, primarily).

**Comment by yathomasi** [comment]
Thanks, @rogermparent.
It looks like we hadn't noticed it earlier https://cml.dev/404.

**Comment by julieg18** [comment]
> It looks like we hadn't noticed it earlier https://cml.dev/404.

Opened an issue! (#238)

**Comment by yathomasi** [comment]
@julieg18 Thanks for opening the issue.
I have pushed changes to fix that as well.

**Pull request merged**